File: backend/app/schemas/retrieval.py
import uuid

from pydantic import BaseModel, Field

from app.models.enums import AuthorityLevel, CollectionType


class SearchRequest(BaseModel):
    query: str = Field(min_length=1)
    top_k: int = Field(default=10, ge=1, le=50)
    collection_types: list[CollectionType] | None = None


class SearchResultChunk(BaseModel):
    chunk_id: uuid.UUID
    document_id: uuid.UUID
    document_title: str
    collection_type: CollectionType
    authority_level: AuthorityLevel | None
    text: str
    citation_label: str
    page_number: int | None
    heading_path: str | None
    score: float

File: frontend/src/types/retrieval.ts
export interface SearchResultChunk {
  chunk_id: string;
  document_id: string;
  document_title: string;
  collection_type: string;
  authority_level: string | null;
  text: string;
  citation_label: string;
  page_number: number | null;
  heading_path: string | null;
  score: number;
}
